filemanager/gcsmanager_test.go

```diff
@@ -0,0 +1,101 @@
+package filemanager
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-go-kit/testhelper"
+
+	"github.com/fsouza/fake-gcs-server/fakestorage"
+)
+
+func TestGCSManagerOpts(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// testcases:
+	tcs := []struct {
+		Name                string
+		GCSUploadIfNotExist bool
+	}{
```

Comment:
Not important though, but we can have these non-exported.
```suggestion
	tcs := []struct {
		name                string
		gcsUploadIfNotExist bool
	}{
```